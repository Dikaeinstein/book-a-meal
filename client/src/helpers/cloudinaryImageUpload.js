import axios from 'axios';

const cloudinaryImageUpload = async (file) => {
  // Make an AJAX upload request using axios
  try {
    // Initialize FormData
    const formData = new FormData();
    formData.append('file', file);
    formData.append('tags', 'meal');
    formData.append('upload_preset', 'a3agftie');
    formData.append('api_key', '553453189171282');
    formData.append('timestamp', parseInt(Date.now() / 1000, 10));
    const headers = { 'X-Requested-With': 'XMLHttpRequest' };
    const { data } = await axios
      .post(
        'https://api.cloudinary.com/v1_1/dikaeinstein/image/upload',
        formData,
        { headers },
      );
    return data.secure_url;
  } catch (error) {
    throw error;
  }
};

export default cloudinaryImageUpload;
