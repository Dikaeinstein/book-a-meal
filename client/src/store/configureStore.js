import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import throttle from 'lodash.throttle';
import { loadState, saveState } from '../helpers/localStorage';
import initialState from '../reducers/initialState';
import rootReducer from '../reducers';

/**
 * Return redux middleware
 */
const getMiddleware = () => {
  const logger = createLogger();

  const middleware =
    process.env.NODE_ENV !== 'production'
    && process.env.NODE_ENV !== 'test'
      ? applyMiddleware(thunk, logger)
      : applyMiddleware(thunk);

  return middleware;
};

/**
 * Configure the redux store
 *
 * @returns {object} Redux store
 */
const configureStore = () => {
  const persistedState = loadState();

  const state = persistedState
    ? { ...initialState, ...persistedState }
    : initialState;

  const store = createStore(
    rootReducer,
    state,
    compose(
      getMiddleware(),
      window.devToolsExtension ? window.devToolsExtension() : empty => empty,
    ),
  );

  // Throttle the saveState function to be called once every second
  store.subscribe(throttle(() => {
    saveState({ orders: store.getState().orders });
  }, 1000));

  return store;
};

export default configureStore;
