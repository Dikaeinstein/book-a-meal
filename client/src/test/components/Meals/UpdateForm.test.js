import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import cloudinaryImageUpload from '../../../helpers/cloudinaryImageUpload';
import ConnectedUpdateMealForm, { UpdateMealForm }
  from '../../../components/Meals/UpdateMealForm';

describe('<UpdateMealForm />', () => {
  const props = {
    errors: {
      updateMeal: null,
      name: null,
      price: null,
      description: null,
    },
    touched: { name: false, price: false, description: false },
    isUpdating: false,
    values: { imageUrl: '' },
  };
  it('should call handleImageChange when image changes', () => {
    const spy = jest.spyOn(UpdateMealForm.prototype, 'handleImageChange');
    const wrapper = shallow(<UpdateMealForm {...props} />);
    cloudinaryImageUpload.mockResolvedValue(Promise.resolve('http://imageurl'));
    let fileInput = wrapper.find('input[type="file"]');
    fileInput.simulate('change', {
      target: {
        files: [{ name: 'mockImage' }],
      },
    });
    expect(spy).toHaveBeenCalled();

    cloudinaryImageUpload.mockResolvedValue(Promise.reject(new Error()));
    fileInput = wrapper.find('input[type="file"]');
    fileInput.simulate('change', {
      target: {
        files: [{ name: 'mockImage' }],
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
  it('should render with correct props from redux', () => {
    const mockStore = configureStore();
    const store = mockStore({ meals: { isUpdating: false } });
    const wrapper = shallow(<ConnectedUpdateMealForm {...props} store={store} />);
    expect(wrapper.props().isUpdating).toEqual(false);
  });
  it('should match snapshot', () => {
    let tree = shallow(<UpdateMealForm {...props} />);
    expect(tree).toMatchSnapshot();

    tree = shallow(<UpdateMealForm {...{ ...props, isUpdating: true }} />);
    expect(tree).toMatchSnapshot();

    tree = shallow(<UpdateMealForm {...{ ...props, errors: { updateMeal: 'true' } }} />);
    expect(tree).toMatchSnapshot();
  });
});
