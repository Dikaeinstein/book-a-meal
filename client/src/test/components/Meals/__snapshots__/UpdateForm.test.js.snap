// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateMealForm /> should call handleImageChange when image changes 1`] = `
<Form
  className="form card"
>
  <div>
    <Label
      className="label label-block"
      htmlFor="name"
    >
      Meal name:
    </Label>
    <Field
      className=""
      name="name"
      placeholder="Enter meal name"
      required={true}
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="price"
    >
      Price (₦):
    </Label>
    <Field
      className=""
      name="price"
      placeholder="Enter price"
      required={true}
      type="number"
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="description"
    >
      Description:
    </Label>
    <Field
      className=""
      component="textarea"
      name="description"
      placeholder="Enter price"
      required={true}
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="image"
    >
      Image:
    </Label>
    <input
      accept="image/*"
      name="image"
      onChange={[Function]}
      type="file"
    />
    <div
      className="text-dark text-center uploading"
    >
      <small>
        <Loading
          text="uploading image . . ."
        >
          <ProgressBar />
        </Loading>
      </small>
    </div>
  </div>
  <div>
    <Field
      name="imageUrl"
      type="hidden"
    />
  </div>
  <Button
    className="btn btn-default font-weight-bold"
    disabled={false}
    type="submit"
    value="update meal"
  />
</Form>
`;

exports[`<UpdateMealForm /> should match snapshot 1`] = `
<Form
  className="form card"
>
  <div>
    <Label
      className="label label-block"
      htmlFor="name"
    >
      Meal name:
    </Label>
    <Field
      className=""
      name="name"
      placeholder="Enter meal name"
      required={true}
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="price"
    >
      Price (₦):
    </Label>
    <Field
      className=""
      name="price"
      placeholder="Enter price"
      required={true}
      type="number"
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="description"
    >
      Description:
    </Label>
    <Field
      className=""
      component="textarea"
      name="description"
      placeholder="Enter price"
      required={true}
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="image"
    >
      Image:
    </Label>
    <input
      accept="image/*"
      name="image"
      onChange={[Function]}
      type="file"
    />
  </div>
  <div>
    <Field
      name="imageUrl"
      type="hidden"
    />
  </div>
  <Button
    className="btn btn-default font-weight-bold"
    disabled={false}
    type="submit"
    value="update meal"
  />
</Form>
`;

exports[`<UpdateMealForm /> should match snapshot 2`] = `
<Form
  className="form card"
>
  <div>
    <Label
      className="label label-block"
      htmlFor="name"
    >
      Meal name:
    </Label>
    <Field
      className=""
      name="name"
      placeholder="Enter meal name"
      required={true}
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="price"
    >
      Price (₦):
    </Label>
    <Field
      className=""
      name="price"
      placeholder="Enter price"
      required={true}
      type="number"
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="description"
    >
      Description:
    </Label>
    <Field
      className=""
      component="textarea"
      name="description"
      placeholder="Enter price"
      required={true}
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="image"
    >
      Image:
    </Label>
    <input
      accept="image/*"
      name="image"
      onChange={[Function]}
      type="file"
    />
  </div>
  <div>
    <Field
      name="imageUrl"
      type="hidden"
    />
  </div>
  <Button
    className="btn btn-default font-weight-bold"
    disabled={true}
    type="submit"
    value="update meal"
  />
  <Loading
    text="Saving Meal . . ."
  >
    <ProgressBar />
  </Loading>
</Form>
`;

exports[`<UpdateMealForm /> should match snapshot 3`] = `
<Form
  className="form card"
>
  <div>
    <Label
      className="label label-block"
      htmlFor="name"
    >
      Meal name:
    </Label>
    <Field
      className=""
      name="name"
      placeholder="Enter meal name"
      required={true}
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="price"
    >
      Price (₦):
    </Label>
    <Field
      className=""
      name="price"
      placeholder="Enter price"
      required={true}
      type="number"
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="description"
    >
      Description:
    </Label>
    <Field
      className=""
      component="textarea"
      name="description"
      placeholder="Enter price"
      required={true}
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="image"
    >
      Image:
    </Label>
    <input
      accept="image/*"
      name="image"
      onChange={[Function]}
      type="file"
    />
  </div>
  <div>
    <Field
      name="imageUrl"
      type="hidden"
    />
  </div>
  <Button
    className="btn btn-default font-weight-bold"
    disabled={false}
    type="submit"
    value="update meal"
  />
  <div
    className="error text-center"
  >
    true
  </div>
</Form>
`;
